daemon/c_net.c

```diff
@@ -1183,6 +1458,24 @@ c_net_cleanup(c_net_t *net, bool is_rebooting)
 		net->fd_netns = -1;
 	}
 
+	/* remove phys network intrefaces from container */
+	uint8_t if_mac[6];
+	for (list_t *l = net->pnet_mv_list; l; l = l->next) {
+		container_pnet_cfg_t *cfg = l->data;
+		char *if_name = (network_str_to_mac_addr(cfg->pnet_name, if_mac) != -1) ?
+					network_get_ifname_by_addr_new(if_mac) :
+					mem_strdup(cfg->pnet_name);
+		if (!cfg->mac_filter) { // skip directly moved if will fallback to rootns
+			continue;
```

Comment:
mem_free(if_name) is missing here

Reply:
Fixed: moved the mem allocation into else block!